pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -873,14 +872,6 @@ def register_plugins(cls, plugin_registry: Dict) -> None:
             offload_optimizer_device="nvme",
         )
 
-    @property
-    def checkpoint_io(self) -> CheckpointIO:
```

Comment:
Just noticed that DeepSpeed overrides `save_checkpoint` and `load_checkpoint`, but not `remove_checkpoint`. Is that correct? Will the removal work with `TorchCheckpointIO`?

Reply:
Yes, I think so because it has `recursive=True`

Reply:
Might be better to delete the directory directly. This might be cleaner in `TorchCheckpointIO`.

```py
        fs = get_filesystem(path)
        if fs.exists(path):
            if fs.isdir(path):
                fs.rmdir(path)
            else:
                fs.rm(path, recursive=True)
            log.debug(f"Removed checkpoint: {path}")
```

Reply:
Rmdir is only for empty directories.